contracts/swap/swap.go

```diff
@@ -83,6 +81,16 @@ type ChequeResult struct {
 	CashTimeout *big.Int
 }
 
+type CashChequeResult struct {
+	Beneficiary      common.Address
+	Recipient        common.Address
+	Caller           common.Address
+	TotalPayout      *big.Int
```

Comment:
added comments.

`CumulativePayout` is the value in the cheque. `TotalPayout` is what was paid out in this call (e.g. if fully liquid then it would be `CumulativePayout` - `paidOut`). `CallerPayout` is the portion of the payout that goes to the caller of the function, it can be used to incentivise third parties to submit the transaction but we don't use that functionality here. For `CashChequeBeneficiary` calls this will always be 0.